lib/type-annotations.ts

```diff
@@ -0,0 +1,106 @@
+import { Fork } from "../types";
+import typesPlugin, { Field, Type } from "./types";
+import { assertNever } from "./utils";
+
+const Op = Object.prototype;
+const hasOwn = Op.hasOwnProperty;
+
+export interface TSTypeAnnotatorConfig {
+  getReferenceToKind(typeName: string): any;
```

Comment:
It was also unclear to me what boundary we were trying to draw bringing in only part of the type generation into lib and leaving other parts in gen-types.ts here https://github.com/benjamn/ast-types/pull/300/commits/c6daab881535a26918bb3aeb01f85e91c0684ca0

Can you explain a bit more why this code should be in lib vs it all being in gen-types?

Reply:
You're right, it was awkward for the `gen-types.ts` code that was generating the type declarations to have to call `field.type.getTypeAnnotation()`, blindly trusting some arbitrary code from another module. I think that division was just an artifact/accident of object-oriented style, since I felt like the individual type classes (`ArrayType`, `OrType`, etc.) should only be responsible for generating the type annotations that they represented. Moving all this logic into `gen-types.ts` makes sense.